ui/js/models/resultsets_store.js

```diff
@@ -786,6 +810,29 @@ treeherder.factory('ThResultSetStore', [
             return repositories[repoName].rsMap[resultsetId].rs_obj;
         };
 
+        var getSelectedRunnableJobs = function(repoName, resultsetId){
+            if (!repositories[repoName].rsMap[resultsetId].selected_runnable_jobs)
+                repositories[repoName].rsMap[resultsetId].selected_runnable_jobs = [];
+            return repositories[repoName].rsMap[resultsetId].selected_runnable_jobs;
+        };
+
+        var toggleSelectedRunnableJob = function(repoName, resultsetId, buildername){
+            var selectedRunnableJobs = getSelectedRunnableJobs(repoName, resultsetId);
+            var jobIndex = selectedRunnableJobs.indexOf(buildername);
+
+            if (jobIndex === -1){
+                selectedRunnableJobs.push(buildername);
+            } else {
+                selectedRunnableJobs.splice(jobIndex, 1);
+            }
+        };
+
+        var isRunnableJobSelected = function(repoName, resultsetId, buildername){
+            var selectedRunnableJobs = getSelectedRunnableJobs(repoName, resultsetId);
```

Comment:
not a big deal, but you can use this function for stuff like this: https://lodash.com/docs#includes